corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -141,6 +141,11 @@ hqDefine("cloudcare/js/formplayer/app", function () {
         }
     });
 
+    FormplayerFrontend.on('showInputError', function (errorMessage, inputId) {
+        let jQueryFriendlyId = "#" + inputId.replace(" ", "\\ ");
```

Comment:
This feels fragile - I didn't realize we used the user-provided input names as IDs, and at a glance it looks like we do that only sometimes (select, multiselect, and geocoder questions don't do that).  It's especially odd because I see geocoder questions use the model `id`, which sounds like the right approach to use generally (though I'm not familiar with where that comes from, so maybe not)

Instead, I think you can pass in the actual input element here, since this is called code that already has access to it.  Something like this should work to get it (in `query.js`, where `trigger` is called):
```js
$(childView.el).find('.query-input-group')
```
It'd be nice to do this in a more "backbone" way, like how `hasError` works, though I'm not certain how plausible that is with this `showError` construct.

Reply:
I totally agree, it does seem like a not-ideal way of assigning id.